### ISSUE TITLE
Agoric CLI leaves behind packages moved to endo

### ISSUE THREAD

**Issue by mhofman** [Agoric/Agoric#4626]
## Describe the bug
When an `agoric-sdk` package is renamed or moved to a package to `@endo` (recent example being `@agoric/marshal`), the `agoric-cli` will no longer be aware of that package. Dapps that were created with previous versions of the `agoric` CLI will continue to depend on the old package name.

In the case of a linked `agoric-sdk`, the package will now likely be pulled from whatever version existed on NPM at the time.  When installing an explicit version like `dev` or `beta` instead of linking, the outdated dependency will similarly be left intact.

## To Reproduce
1. Scaffold a dapp with an old SDK
2. Update the sdk
3. `agoric install` or `agoric install dev`

## Expected behavior
The cli should rewrite outdated dependencies to their new name, and warn the developer to update their code.

The question remains as to what version of those dependencies should be used. One approach would be to bundle in the CLI source the same version number that other agoric-sdk packages depend on. Maybe endo's `sync-version` could be (ab)used to that effect?

## Additional context
This came up while thinking about https://github.com/Agoric/agoric-sdk/issues/4441

We may want to use a more explicit `upgrade` command if this condition arises and fail the normal `install` if the agoric-sdk version is incompatible.

**Comment by michaelfig** [comment]
At the very least, we should:
```
npm deprecate "@agoric/some-package@<= latest-version" "Moved to @endo/some-package"
```

As for the right thing to do for an upgrade, I think manual changes is the only thing we can do.  It's now up to every consumer of the `@endo/*` packages to be aware of the package to use and the semver, unless `@endo` adopts a continuous publishing system the way Agoric SDK (very briefly) had.

I wrote `agoric install` for a simpler time, but `@agoric` can no longer be consumed as a monolith.  I think `agoric install` will have to go away, to be replaced by standard tools.

**Comment by michaelfig** [comment]
@kriskowal thoughts on the above?

**Comment by kriskowal** [comment]
@michaelfig I agree we need to at least `npm deprecate` the Agoric packages. I will take the ball on that and use this issue to track until it’s done. I also agree we need in general to move away from `agoric install` workflows and that the next steps will be removing the postinstall script in our scaffold getting started dapp and revise the getting started guide to reflect that new reality.

Dapps need to use Semver for Endo dependencies. The ones I’ve migrated already are. I think the best we can do for contract writers will be to keep a running document with migration instructions for every breaking change in any Agoric SDK package or transitive dependency. If we create any automation, I can’t imagine taking a human out of the loop.

**Comment by kriskowal** [comment]
I’ve marked all of the deprecated packages that have moved to Endo. For example, you should see a banner at https://www.npmjs.com/package/@agoric/captp

**Comment by kriskowal** [comment]
@michaelfig I’ve drafted the work I believe lies on the road to revising the getting started guide to reflect our new versioned agoric-sdk reality https://github.com/Agoric/agoric-sdk/issues/4676
